src/lakefs_spec/client_helpers.py

```diff
@@ -31,59 +32,88 @@ def commit(
         return rev_parse(client, repository, branch, parent=0)
 
-    commit_creation = CommitCreation(message=message, metadata=metadata or {})
+    commit_creation = CommitCreation(message=message, metadata=metadata)
 
     new_commit = client.commits_api.commit(
         repository=repository, branch=branch, commit_creation=commit_creation
     )
     return new_commit
 
 
-def create_tag(client: LakeFSClient, repository: str, ref: str | Commit, tag: str) -> None:
-    if isinstance(ref, Commit):
-        ref = ref.id
-    tag_creation = TagCreation(id=tag, ref=ref)
-    client.tags_api.create_tag(repository=repository, tag_creation=tag_creation)
-
-
-def ensure_branch(client: LakeFSClient, repository: str, branch: str, source_branch: str) -> str:
+def create_branch(
+    client: LakeFSClient, repository: str, name: str, source_branch: str, exist_ok: bool = True
+) -> str:
     """
-    Creates a branch named ``branch`` if not already existent.
+    Create a branch in a lakeFS repository.
 
     Parameters
     ----------
     client: LakeFSClient
         The lakeFS client configured for (and authenticated with) the target instance.
     repository: str
         Repository name.
-    branch: str
-        Name of the branch.
+    name: str
+        Name of the newly created branch.
     source_branch: str
         Name of the source branch the new branch is created from.
+    exist_ok: bool
+        Ignore errors if the branch already exists.
 
     Returns
     -------
-    The branch name that was given.
+    The newly created branch name.
     """
 
     try:
-        new_branch = BranchCreation(name=branch, source=source_branch)
-        # client.branches_api.create_branch throws ApiException when branch exists
+        new_branch = BranchCreation(name=name, source=source_branch)
+        # client.branches_api.create_branch throws ApiException if branch exists
         client.branches_api.create_branch(repository=repository, branch_creation=new_branch)
-        logger.info(f"Created new branch {branch!r} from branch {source_branch!r}.")
-    except ApiException:
-        pass
+        logger.debug(f"Created new branch {name!r} from branch {source_branch!r}.")
+        return name
+    except ApiException as e:
+        if e.status == 409 and exist_ok:
+            return name
+        raise e
+
 
-    return branch
+def create_repository(
+    client: LakeFSClient, name: str, storage_namespace: str, exist_ok: bool = True
+) -> Repository:
+    """
+    Create a repository with the given name and storage namespace.
 
+    Important: Due to cleanup issues in the lakeFS backend, creating a repository again after prior
+    deletion under the same name and storage namespace will almost certainly not work.
 
-def get_tags(client: LakeFSClient, repository: str) -> dict:
-    return client.tags_api.list_tags(repository=repository)
+    The exist_ok flag only asserts idempotence in case the repository is not deleted between
```

Comment:
```suggestion
    The `exist_ok` flag only asserts idempotence in case the repository is not deleted between
```

Reply:
Addressed in [643d821](https://github.com/aai-institute/lakefs-spec/pull/154/commits/643d821ad647f440a4dca82841dcbcd65d9c79b7).